src/common/src/array/arrow.rs

```diff
@@ -17,9 +17,37 @@ use arrow_schema::Field;
 use chrono::{NaiveDateTime, NaiveTime};
 use itertools::Itertools;
 
+use super::column::Column;
 use super::*;
 use crate::types::struct_type::StructType;
 
+// Implement bi-directional `From` between `DataChunk` and `arrow_array::RecordBatch`.
+
+impl From<&DataChunk> for arrow_array::RecordBatch {
```

Comment:
`TryFrom`? 🤔 (BTW, seems not used)

Reply:
According to the [document](https://docs.rs/arrow-array/30.0.1/arrow_array/struct.RecordBatch.html#method.try_new) from arrow, this function should not fail as long as we provide a valid `DataChunk`. So let's keep this `From` until we find other exceptions.

It is not used by this PR, but I think it might be useful someday.